plugins/modules/clustering/nomad/nomad_job.py

```diff
@@ -0,0 +1,246 @@
+version_added: "1.2.0"
+short_description: Launch an Nomad Job
+description:
+    - Launch an Nomad job. See
+      U(https://www.nomadproject.io/api-docs/jobs/) for an overview.
+options:
+    host:
+      description:
+        - FQDN of nomad server.
+      required: true
+      type: str
+    secure:
+      description:
+        - TLS/SSL connection.
+      type: bool
+      default: false
+    timeout:
+      description:
+        - Timeout (in seconds) for the request to nomad.
+      type: int
+      default: 5
+    validate_certs:
+      description:
+        - Enable TLS/SSL certificate validation.
+      type: bool
+      default: true
+    cert:
+      description:
+        - Path of certificate TLS/SSL.
+      type: path
+    key:
+      description:
+        - Path of certificate key TLS/SSL.
+      type: path
+    namespace:
+      description:
+        - Namespace for nomad.
+      type: str
+    token:
+      description:
+        - ACL token for authentification.
+      type: str
+    name:
+      description:
+        - Name of job for delete,stop and start job without source.
+      type: str
+    state:
+      description:
+        - Deploy or remove job.
+      choices: ["present", "absent"]
+      required: true
+      type: str
+    force_start:
+      description:
+        - Force job to started.
+      type: bool
+      default: false
+    source_json:
+      description:
+        - Source of json nomad job.
+      type: json
+    source_hcl:
+      description:
+        - Source of hcl nomad job.
+      type: str
+'''
+
+EXAMPLES = '''
+- name: Create job
+  community.general.nomad_job:
+    host: localhost
+    state: present
+    source_hcl: "{{ lookup('ansible.builtin.file', 'job.hcl') }}"
+    timeout: 120
+
+- name: stop job
+  community.general.nomad_job:
+    host: localhost
+    state: absent
+    name: api
+
+- name: force job to start
+  community.general.nomad_job:
+    host: localhost
+    state: present
+    name: api
+    timeout: 120
+    force_start: true
+'''
+
+
+import os
+import json
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+import_nomad = None
+try:
+    import nomad
+    import_nomad = True
+except ImportError:
+    import_nomad = False
+
+
+def run():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(required=True, type='str'),
+            state=dict(required=True, choices=['present', 'absent']),
+            secure=dict(type='bool', default=False),
+            timeout=dict(type='int', default=5),
+            validate_certs=dict(type='bool', default=True),
+            cert=dict(type='path', default=None),
+            key=dict(type='path', default=None),
+            namespace=dict(type='str', default=None),
+            name=dict(type='str', default=None),
+            source_json=dict(type='json', default=None),
+            source_hcl=dict(type='str', default=None),
```

Comment:
rather than having 2 different sources with types, I would suggest to rename to `content` (a common field name, also used in other modules) and have a `content_type` field with choices `['json','hcl','yaml']` and default it to `hcl`.

Reply:
I agree to @resmo

Reply:
ok